src/opnsense/mvc/app/library/OPNsense/Backup/Base.php

```diff
@@ -35,21 +35,30 @@
  */
 abstract class Base
 {
+    /**
+     * @var string[] openssl arguments by encryption method
+     */
+    private $openssl_arguments = [
+        "M0" => "-aes-256-cbc -md md5", /* backwards-compatible default */
+        "M1" => "-aes-256-cbc -pbkdf2 -iter 100000 -md SHA512",
```

Comment:
CBC is still a good idea? We could just bump hash and iterate from there instead of changing all at once.

Reply:
To me it looks we could stick with CBC here. AFAIK the only CBC vulnerability is the [padding oracle attack](https://blog.cloudflare.com/padding-oracles-and-the-decline-of-cbc-mode-ciphersuites/), which does not seem to be relevant here as it requires an attacker to inject plain text.

Reply:
Ah, and AES(256) (CBC) is still listed as a [recommended strong cipher suite on bettercrypto.org](https://bettercrypto.org/#_configuration_a_strong_ciphers_fewer_clients).